app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java

```diff
@@ -1,45 +1,120 @@
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
 
         final View v = inflater.inflate(R.layout.fragment_alarm_landing_page, container, false);
-
+        vibrator = (Vibrator) getActivity().getSystemService(VIBRATOR_SERVICE);
+        sensorManager = (SensorManager) getActivity().getSystemService(Context.SENSOR_SERVICE);
+        accelerormeterSensor = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
         final Button launchMainActivityBtn = (Button) v.findViewById(R.id.load_main_activity_btn);
         final Button dismiss = (Button) v.findViewById(R.id.dismiss_btn);
+        launchMainActivityBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startActivity(new Intent(getContext(), MainActivity.class));
+                vibrator.cancel();
+                getActivity().finish();
+            }
+        });
+        dismiss.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                vibrator.cancel();
+                getActivity().finish();
+            }
+        });
+        pm=(PowerManager)getActivity().getSystemService(Context.POWER_SERVICE);
+        //wl=pm.newWakeLock(PowerManager.SCREEN_BRIGHT_WAKE_LOCK,"mytag");
 
-        launchMainActivityBtn.setOnClickListener(this);
-        dismiss.setOnClickListener(this);
 
+        startvibe();
         return v;
     }
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (sensorManager != null)
+            sensorManager.unregisterListener(this);
+        // wl.release();
+    }
 
     @Override
-    public void onClick(View view) {
+    public void onAccuracyChanged(Sensor sensor, int accuracy) {
 
-        switch (view.getId()) {
-            case R.id.load_main_activity_btn:
-                startActivity(new Intent(getContext(), MainActivity.class));
-                getActivity().finish();
-                break;
-            case R.id.dismiss_btn:
-                getActivity().finish();
-                break;
+    }
+    @Override
+    public void onSensorChanged(SensorEvent event) {
+        if (event.sensor.getType() == Sensor.TYPE_ACCELEROMETER) {
+            long currentTime = System.currentTimeMillis();
+            long gabOfTime = (currentTime - lastTime);
+            if (gabOfTime > 100) {
+                lastTime = currentTime;
+                x = event.values[SensorManager.DATA_X];
+                y = event.values[SensorManager.DATA_Y];
+                z = event.values[SensorManager.DATA_Z];
+                speed = Math.abs(x + y + z - lastX - lastY - lastZ) / gabOfTime * 10000;
+                if (speed > SHAKE_THRESHOLD) {
+                    lastTime=currentTime;
+                    count++;//for shake test
+                    if(count==10)
```

Comment:
Not for this PR, but after I've merged in your work I will probably make these optional, configurable features (vibrating main activity, shake to dismiss, vibration pattern, number of shakes required to dismiss, etc.)

Reply:
@PPartisan It's okay to tweak the functionality a bit to suit your specific characteristics.